### PR TITLE
Fix comments

### LINKED ISSUE #53: Single quote in inline comment breaks highlighting for the rest of the file
screen shot 2015-03-03 at 2 01 06 pm

### LINKED ISSUE #82: Relative protocol highlights as if it were commented out
```css
.brand {
    background: transparent url(//placehold.it/350x500) 0 0 / cover no-repeat;
}
```

It renders completely fine in the output css it just appears to be commented in the editor.

screen shot 2015-10-22 at 4 21 01 pm

screen shot 2015-10-22 at 4 22 30 pm

Perhaps there's a way to detect if // is within url() and disable the comment highlighting?

### PR DESCRIPTION
**Pull request by 50Wliu**
The first commit fixes an issue in SCSS where URLs would be tokenized as comments.
The second commit fixes an issue in SASS where single-line comments were being tokenized incorrectly.

Fixes #82
Fixes #53

### CODE REVIEW

**Comment by kevinsawicki** [comment]
#87 has now been merged

**Comment by 50Wliu** [comment]
Ok.  Now I'm stuck on an issue where whitespace preceding the comment is still being tokenized as a comment.

For example, the following:
```md
    // this is a comment
^^^^ This all gets tokenized as `comment.line.sass` as well
```

Not sure how to fix that since the begin regex must make sure that only whitespace precedes the start of the comment, and then that whitespace match is referenced in the end regex.  I tried using `contentName` and lookaheads/behinds, but apparently backreferences aren't allowed in lookbehinds.

**Comment by MaximSokolov** [comment]
> begin regex must make sure that only whitespace precedes the start of the comment

What if non-whitespace character precedes the comment? Can such comment be tokenized as `invalid.illegal` first to remove `^(\\s*)` from pattern?

**Comment by 50Wliu** [comment]
@MaximSokolov so do you mean something like:
`\\w+\\s*//` -> `invalid.illegal.comment`?

**Comment by MaximSokolov** [comment]
@50Wliu That's right. Just `invalid.illegal`~~.comment~~ (it would be highlighted as `comment` if syntax theme doesn't have rule for `invalid.illegal`)

**Comment by 50Wliu** [comment]
Ok I just pushed a change that makes specs pass.  However, the double matching feels really hacky.  Maybe we should just go with how language-javascript does it (by tokenizing the preceding whitespace as `punctuation.definition.leading.whitespace` or something like that)?

**Comment by MaximSokolov** [comment]
> Maybe we should just go with how language-javascript does it

Whitespaces should be matched in comment pattern to known indentation size (`\\1` in `^(?!\\s{2}\\1)`)  I've tried:
```coffee
{
  'begin': '^(\\s*)(?=//)'
  'end': '(?!\\G)'
  'patterns': [
    {
      'begin': '(//)'
      'beginCaptures':
        '1': 'name': 'punctuation.definition.comment.sass'
      'end': '^(?=\\s?\\S)'
      'name': 'comment.line.sass'
    }
  ]
}
```
```sass
// comment
  comment
not a comment, ok

  // comment
    comment
  comment, wrong
```
I suggest to revert `contentName` -> `name`.
IMO two groups for `//` and comment content is worse than whitespaces as a `comment`

File: grammars/sass.cson [hunk 1]
```diff
@@ -310,13 +310,40 @@
...
-        'name': 'punctuation.definition.comment.sass'
-    'end': '^(?!\\1\\s)'
-    'name': 'comment.sass'
+  'comments':
+    'patterns': [
+      {
+        'match': '\\w+\\s*((//|/\\*).*)'
+        'captures':
+          '1':
+            'name': 'invalid.illegal.comment.sass'
```
Comment (MaximSokolov) [inline grammars/sass.cson]: Just `invalid.illegal.sass`

**Comment by 50Wliu** [comment]
@MaximSokolov I've addressed most of your comments; mind seeing if there's any more ways to improve this?

**Comment by MaximSokolov** [comment]
I don't have anything to add. Looks good to me.

**Pull request merged**